src/Welcome.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import getToken from './lib/tokens';
+import { useHistory } from 'react-router-dom';
+
+const Welcome = () => {
+  let history = useHistory();
+
+  const storeToken = () => {
+    const token = getToken();
+    localStorage.setItem('userToken', token);
+    history.push('/list');
+  };
+  return (
+    <div>
+      <h2>Welcome to your Smart Shopping List!</h2>
+      <button onClick={() => storeToken()}>Create a new list</button>
+    </div>
+  );
+};
+
```

Comment:
Definitely not a change request, but a different way you could approach this would be to just have this component receive a prop called onClick, and then it might look like this:

`const Welcome = ({onClick}) => `

Then the `storeToken` function wouldn't exist here.

and in the button it would just be `onClick={onClick}`

Reply:
We actually tried passing the function as a prop to `Welcome` component, but then decided that it would be more straightforward to have it within the component. I would love to discuss the pros and cons of both approaches tomorrow!